### PR TITLE
Resource_container_cluster : check node_version and min_master_version on create

### PR DESCRIPTION
**Pull request by Edwinhr716**
Resource_container_cluster : check node_version and min_master_version on create when calling `terraform plan`

Fixes https://github.com/hashicorp/terraform-provider-google/issues/2701

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
container: added check that node_version and min_master_version are the same on create, when running terraform plan
```

### CODE REVIEW

**Comment by melinath** [comment]
@Edwinhr716 at a glance this seems reasonable but I'll need to more thoroughly review tomorrow - in the meantime could you fix the link to the issue as described in https://googlecloudplatform.github.io/magic-modules/contribute/create-pr/ and fix the release note as described in https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/? Thanks!

**Comment by Edwinhr716** [comment]
> @Edwinhr716 at a glance this seems reasonable but I'll need to more thoroughly review tomorrow - in the meantime could you fix the link to the issue as described in https://googlecloudplatform.github.io/magic-modules/contribute/create-pr/ and fix the release note as described in https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/? Thanks!

Done

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -6395,3 +6396,27 @@ func containerClusterEnableK8sBetaApisCustomizeDiffFunc(d tpgresource.TerraformR
 
 	return nil
 }
+
+func containerClusterNodeVersionCustomizeDiff(_ context.Context,diff *schema.ResourceDiff, meta interface{}) error {
```
Comment (melinath) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:6400]: Please add thorough unit tests for this customize diff func - similar to https://github.com/hashicorp/terraform-provider-google-beta/blob/dd55378df33eeea24904814063280c6057bb3e45/google-beta/services/compute/resource_compute_instance_template_internal_test.go#L156

**Review by melinath — CHANGES_REQUESTED** [review]
Thanks for updating the initial comment - looks good! The code LGTM except that it would be great to have some more tests.

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb (hunk 1 above)
Comment (Edwinhr716) [reply mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:6400]: Added more tests

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 2]
```diff
@@ -6395,3 +6396,32 @@ func containerClusterEnableK8sBetaApisCustomizeDiffFunc(d tpgresource.TerraformR
...
+func containerClusterNodeVersionCustomizeDiffFunc(diff tpgresource.TerraformResourceDiff) error {
+	oldValueName, _ := diff.GetChange("name")
+	if oldValueName != "" {
+		return nil
+	}
+
+	_, newValueNode := diff.GetChange("node_version")
+	_, newValueMaster := diff.GetChange("min_master_version")
+
+	if newValueNode == "" || newValueMaster == "" {
```
Comment (melinath) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:6414]: could you add a test for each of these cases?

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 3]
```diff
@@ -6395,3 +6396,32 @@ func containerClusterEnableK8sBetaApisCustomizeDiffFunc(d tpgresource.TerraformR
...
+		return nil
+	}
+
+	//ignore -gke.X suffix for now. If it becomes a problem later, we can fix it
```
Comment (melinath) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:6418]: It looks like cluster versions don't necessarily include a `-gke.X` suffix - could you add at least one test case for each field to cover that? https://cloud.google.com/kubernetes-engine/versioning#specifying_cluster_version

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 4]
```diff
@@ -6395,3 +6396,32 @@ func containerClusterEnableK8sBetaApisCustomizeDiffFunc(d tpgresource.TerraformR
...
 }
+
+func containerClusterNodeVersionCustomizeDiff(_ context.Context,diff *schema.ResourceDiff, meta interface{}) error {
+	// separate func to allow unit testing
+	return containerClusterNodeVersionCustomizeDiffFunc(diff)
+}
+
+func containerClusterNodeVersionCustomizeDiffFunc(diff tpgresource.TerraformResourceDiff) error {
+	oldValueName, _ := diff.GetChange("name")
+	if oldValueName != "" {
```
Comment (melinath) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:6407]: could you add a test case to cover this?

**Review by melinath — CHANGES_REQUESTED** [review]
a few more test cases that it would be great to add to the unit tests!

**Review by melinath — APPROVED** [review]
LGTM

**Pull request merged**